### PR TITLE
out_s3: miscellaneous fixes

### PR DESCRIPTION
**Pull request by PettitWesley**
----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

**Documentation**

- [ ] Documentation required for this feature

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Review by fujimotos — APPROVED** [review]

**Comment by edsiper** [comment]
Thanks 

Please adjust commit message for 401bd2d

**Comment by fujimotos** [comment]
@eduardo FWIW, With this patch and the following patches applied,
I can confirm that [the master HEAD is compilable on Windows.](https://ci.appveyor.com/project/fujimotos/fluent-bit-github/builds/35573546)

 * https://github.com/edsiper/chunkio/pull/53 (applied to lib/chunkio)
 * https://github.com/fluent/fluent-bit/pull/2631
 * https://github.com/fluent/fluent-bit/pull/2632

File: plugins/out_s3/s3_store.c [hunk 1]
```diff
@@ -212,7 +214,7 @@ int s3_store_init(struct flb_s3 *ctx)
     struct flb_fstore_stream *fs_stream;
 
     /* Initialize the storage context */
-    fs = flb_fstore_create(ctx->store_dir);
```
Comment (edsiper) [inline plugins/out_s3/s3_store.c:215]: hmm buffer dir is there to include the bucket name in the filesystem path, but we are not longer using it. Is this working 100% working ?, I am not confident backlog chunks will be loaded if there is a different directory structure

**Comment by edsiper** [comment]
thanks @PettitWesley and @fujimotos 

I did a manual pick of the commits on this PR 

note: PR MANUALLY MERGED

File: plugins/out_s3/s3_store.c (hunk 1 above)
Comment (PettitWesley) [reply plugins/out_s3/s3_store.c:215]: It works

**Pull request closed without merging**